src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -232,32 +342,66 @@ public ValueTask DisposeAsync()
         return _app.DisposeAsync();
     }
 
-    private static void ConfigureListenAddresses(KestrelServerOptions kestrelOptions, Uri[] uris, Action<ListenOptions, bool>? configureListenOptions = null)
+    private static bool IsHttps(Uri uri) => string.Equals(uri.Scheme, "https", StringComparison.Ordinal);
+
+    private static DashboardConfiguration LoadDashboardConfig(IConfiguration configuration)
     {
-        foreach (var uri in uris)
+        var browserUris = configuration.GetUris(DashboardUrlVariableName, new(DashboardUrlDefaultValue));
+        if (browserUris.Length == 0)
         {
-            if (string.Equals(uri.Host, "localhost", StringComparisons.UrlHost))
-            {
-                kestrelOptions.ListenLocalhost(uri.Port, ConfigureListenOptions);
-            }
-            else
+            throw new InvalidOperationException($"One URL for Aspire dashboard OTLP endpoint with {DashboardUrlDefaultValue} is required.");
+        }
+
+        var otlpUris = configuration.GetUris(DashboardOtlpUrlVariableName, new(DashboardOtlpUrlDefaultValue));
+        if (otlpUris.Length != 1)
+        {
+            throw new InvalidOperationException($"At least one URL for Aspire dashboard browser endpoint with {DashboardOtlpUrlDefaultValue} is required.");
+        }
+
+        OtlpAuthMode otlpAuthMode;
+        string? otlpApiKey = null;
+
+        if (configuration[DashboardOtlpAuthModeVariableName] is { } v)
+        {
+            if (!Enum.TryParse(v, out otlpAuthMode))
             {
-                kestrelOptions.Listen(IPAddress.Parse(uri.Host), uri.Port, ConfigureListenOptions);
+                throw new InvalidOperationException($"Unknown {nameof(OtlpAuthMode)} value: {v}");
             }
 
-            void ConfigureListenOptions(ListenOptions options)
+            switch (otlpAuthMode)
             {
-                var isHttps = IsHttps(uri);
-                if (isHttps)
-                {
-                    options.UseHttps();
-                }
-                configureListenOptions?.Invoke(options, isHttps);
+                case OtlpAuthMode.None:
+                    break;
+                case OtlpAuthMode.ApiKey:
+                    otlpApiKey = configuration[DashboardOtlpApiKeyVariableName];
```

Comment:
Similarly here do we to fork out logic to retreive apiKey for ACA scenario? How would we determine from the appModel whether the dashboard is hosted in ACA? Can be in a separate PR - just curious

Reply:
Still to figure out.